src/Sentry.AspNetCore/SentryMiddleware.cs

```diff
@@ -216,7 +217,9 @@ public async Task InvokeAsync(HttpContext context, RequestDelegate next)
 
             void CaptureException(Exception e, SentryId evtId, string mechanism, string description)
             {
-                e.SetSentryMechanism(mechanism, description, handled: false);
+                var userHandledSet = e.Data.Contains(Mechanism.HandledKey);
+                var userHandledValue = e.Data[Mechanism.HandledKey] as bool?;
+                e.SetSentryMechanism(mechanism, description, handled: userHandledSet ? userHandledValue : false);
```

Comment:
I'm thinking, since we're going to the effort of not overwriting Mechanism.Handled when this has been set by the user, maybe we should also avoid overwriting the Mechanism.Type and Mechanism.Description as well?

Reply:
Good question, I'll take a peek.

Reply:
@jamescrosswell take a look again thanks.